src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -113,7 +113,16 @@ public int HandleDatabaseRecordChange(DatabaseRecord record, long raftIndex)
                 }
                 catch (OperationCanceledException e)
                 {
+                    indexLock.Release();
                     _documentDatabase.RachisLogIndexNotifications.NotifyListenersAbout(raftIndex, e);
+                    try
+                    {
+                        index.Dispose();
```

Comment:
You didn't get a lock and database is being shutdown, which would dispose the index anyway